ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/git/GitServiceClient.java

```diff
@@ -759,6 +759,24 @@ void commit(DevMachine devMachine,
     Promise<Revision> commit(DevMachine devMachine, Path project, String message, Path[] files, boolean amend);
 
     /**
+     * Performs commit changes from index to repository.
+     *
+     * @param devMachine
+     *         current machine
+     * @param project
+     *         project (root of GIT repository)
+     * @param message
+     *         commit log message
+     * @param all
+     *         automatically stage files that have been modified and deleted
+     * @param files
+     *         the list of files that are committed, ignoring the index
+     * @param amend
+     *         indicates that previous commit must be overwritten
+     */
+    Promise<Revision> commit(DevMachine devMachine, Path project, String message, boolean all, Path[] files, boolean amend);
```

Comment:
`all` parameter means add to index all changes and `files` parameter defines what files to commit. `all` can be `true` and `files` parameter can contain items, this will happen when user would call commit operation on selected items and would select `Add all changes except of new files` check box.
We have separate issue: https://github.com/eclipse/che/issues/3614 to improve this window where add to index action will be performed automaticaly because user will have an ability to chose all changed files directly in the window.